src/frontend/src/handler/create_source.rs

```diff
@@ -854,6 +921,7 @@ pub async fn handle_create_source(
         &mut col_id_gen,
         create_source_type,
         overwrite_options.source_rate_limit,
+        SqlColumnStrategy::Reject,
```

Comment:
> Cases 3 and 4 adhere to the current behavior. However, we can now modify this behavior simply by passing a different SqlColumnStrategy, which will accomplish the objective in #12199.

IIUC, the user-facing part will be in future PRs?

Reply:
Yes